Invalid use of incomplete type ‘struct shogun::TParameter’

Latest develop, building on Ubuntu 16.04

 Building CXX object src/shogun/CMakeFiles/libshogun.dir/base/class_list.cpp.o
FAILED: /usr/bin/ccache /usr/bin/c++   -I../src -I../src/shogun -Isrc -Isrc/shogun -isystem /usr/include/eigen3 -I/usr/include/atlas -I/usr/include/libxml2 -I/usr/include/hdf5/serial -std=c++11 -Wall -Wno-unused-parameter -Wformat -Wformat-security -Wparentheses -Wshadow -Wno-unknown-pragmas -Wno-deprecated   -fopenmp   -O3 -fexpensive-optimizations -frerun-cse-after-loop -fcse-follow-jumps -finline-functions -fschedule-insns2 -fthread-jumps -fforce-addr -fstrength-reduce -funroll-loops -mfpmath=sse -fPIC -MMD -MT src/shogun/CMakeFiles/libshogun.dir/base/class_list.cpp.o -MF src/shogun/CMakeFiles/libshogun.dir/base/class_list.cpp.o.d -o src/shogun/CMakeFiles/libshogun.dir/base/class_list.cpp.o -c ../src/shogun/base/class_list.cpp
In file included from ../src/shogun/base/class_list.cpp:15:0:
../src/shogun/evaluation/GradientResult.h: In member function ‘virtual void shogun::CGradientResult::print_result()’:
../src/shogun/evaluation/GradientResult.h:103:42: error: invalid use of incomplete type ‘struct shogun::TParameter’
    const char* param_name=param_node->key->m_name;
                                          ^
In file included from ../src/shogun/evaluation/EvaluationResult.h:15:0,
                 from ../src/shogun/evaluation/GradientResult.h:16,
                 from ../src/shogun/base/class_list.cpp:15:
../src/shogun/base/SGObject.h:38:8: note: forward declaration of ‘struct shogun::TParameter’
 struct TParameter;
        ^

--- a/src/shogun/evaluation/GradientResult.h
+++ b/src/shogun/evaluation/GradientResult.h
@@ -15,6 +15,7 @@
 
 #include <shogun/evaluation/EvaluationResult.h>
 #include <shogun/lib/Map.h>
+#include <shogun/base/Parameter.h>
 
 namespace shogun
 {

